x-pack/plugins/uptime/server/lib/adapters/telemetry/kibana_telemetry_adapter.ts

```diff
@@ -76,10 +73,10 @@ export class KibanaTelemetryAdapter {
           },
         },
       },
-      fetch: async ({ callCluster }: CollectorFetchContext) => {
+      fetch: async ({ esClient }: CollectorFetchContext) => {
         const savedObjectsClient = getSavedObjectsClient()!;
         if (savedObjectsClient) {
-          await this.countNoOfUniqueMonitorAndLocations(callCluster, savedObjectsClient);
+          await this.countNoOfUniqueMonitorAndLocations(esClient, savedObjectsClient);
```

Comment:
```suggestion
    	  const uptimeEsClient = createUptimeESClient({ esClient, savedObjectsClient });
          await this.countNoOfUniqueMonitorAndLocations(uptimeEsClient, savedObjectsClient);
```

i think doing it like this will simplify things

Reply:
@shahzad31 The types for the SavedObjectsClient aren't the same in the collector and in `createUptimeEsClient`. In the collector, `savedObjectsClient` is of type `ISavedObjectsRepository | undefined`, whereas `createUptimeEsClient` expects `savedObjectsClient` to be of type `SavedObjectsClientContract`. We're missing `errors` in the type that's being passed in.

Reply:
@shahzad31 we can get around this by changing the `savedObjectsClient` type expected in `createUptimeESClient` to be `SavedObjectsClientContract | ISavedObjectsRepository`.

Reply:
@TinaHeiligers yeah i think that can work.